src/Sparrow/Hashing.cs

```diff
@@ -511,26 +511,67 @@ public static ulong Calculate(int[] buf, int len = -1, ulong seed = 0)
 
         public static class HashCombiner
         {
+            /// <summary>
+            /// The combine function will perform the mixing of 2 hash values into a single one.
+            /// It is important that the parameters are hash functions. If they are not the statistical properties of the output will be quite bad.
+            /// For non hashes use <see cref="Hashing.Combine"/> instead
+            /// </summary>
+            /// <param name="x">The first hash value</param>
+            /// <param name="y">The second hash value</param>
+            /// <returns>The combined hash value</returns>
             public static int Combine(int x, int y)
             {
                 return CombineInline(x, y);
             }
 
+            /// <summary>
+            /// The combine function will perform the mixing of 2 hash values into a single one.
+            /// It is important that the parameters are hash functions. If they are not the statistical properties of the output will be quite bad.
+            /// For non hashes use <see cref="Hashing.Combine"/> instead
+            /// </summary>
+            /// <param name="x">The first hash value</param>
+            /// <param name="y">The second hash value</param>
+            /// <returns>The combined hash value</returns>
             public static uint Combine(uint x, uint y)
             {
                 return CombineInline(x, y);
             }
 
+            /// <summary>
+            /// The combine function will perform the mixing of 2 hash values into a single one.
+            /// It is important that the parameters are hash functions. If they are not the statistical properties of the output will be quite bad.
+            /// For non hashes use <see cref="Hashing.Combine"/> instead
+            /// </summary>
+            /// <param name="x">The first hash value</param>
+            /// <param name="y">The second hash value</param>
+            /// <returns>The combined hash value</returns>
             public static long Combine(long x, long y)
             {
                 return CombineInline(x, y);
             }
 
+            /// <summary>
+            /// The combine function will perform the mixing of 2 hash values into a single one.
+            /// It is important that the parameters are hash functions. If they are not the statistical properties of the output will be quite bad.
+            /// For non hashes use <see cref="Hashing.Combine"/> instead
```

Comment:
Nope, those are Hashing.HashCombiner.Combine. They are absurdly fast, but values have to be hashes themselves. Hashing.Combine are general purpose but slower.